Fix annotation input focus trap regression in Safari

CetinSert opened the pull request:
`setSelectionRange(0, 0)` added in https://github.com/mozilla/pdf.js/commit/44b24fcc2936e1d75c249aa161970ab8fa83e4d2 for #12359, required only by Firefox ([bug](https://bugzilla.mozilla.org/show_bug.cgi?id=860329)), causes issues mozilla#13191, mozilla#12592 in Safari.
`scrollLeft = 0` is a fix that breaks the focus trap in Safari while **keeping Firefox behavior same for #12359**.

@calixteman

CetinSert committed:
Fixes #13191 and #12592, ...

This causes issues on Safari (see #12592 and #13191). Just removing the line fixes the issues and no difference / regression in behavior was observed without the `event.target.setSelection(0, 0);` bit in other browsers (including Firefox). The blur (switching focus out of the control) does effectively achieve the same thing in all browsers I have tested on.

Tested specifically in Firefox with

1. inspect a `TextWidgetAnnotation` `input` element
2. `$0.setSelectionRange = (a, b) => console.warn(a, b);`
3. type `wow` and select `wow` in the input element
4. click/tab/tap outside the input element
5. <kbd>CTRL</kbd> + <kbd>C</kbd> ➡ no `wow` ✔ copied
6. <kbd>CTRL</kbd> + <kbd>V</kbd> ➡ confirmed

CetinSert committed:
Fixes issue in Safari, no regression in Firefox

setSelectionRange(0, 0); // required only by Firefox, causes issues in Safari (see #12592 and #13191).
scrollLeft = 0; // is a fix that breaks the focus trap in Safari while keeping Firefox behavior same.

CetinSert committed:
Fixes issues in Safari, no regression in Firefox

setSelectionRange(0, 0); // required only by Firefox, causes issues in Safari (see mozilla#12592 and mozilla#13191).
scrollLeft = 0; // is a fix that breaks the focus trap in Safari while keeping Firefox behavior same for #12359

cetinsert commented:
@Snuffleupagus - please see files changed: 1 LoC. Sorry for yesterday, I honestly failed to see the link to #12359.
Minimal case here: http://jsfiddle.net/mv08ekay/4/ (10 LoC - no PDF.js)

Answer from:

- https://stackoverflow.com/questions/67055218/set-input-caret-position-on-blur-in-safari-without-refocusing-input
- https://bugzilla.mozilla.org/show_bug.cgi?id=860329

Safari bug report for `setSelectionRange()`:

- https://bugs.webkit.org/show_bug.cgi?id=224425

Snuffleupagus commented on src/display/annotation_layer.js:
```diff
@@ -668,7 +668,7 @@ class TextWidgetAnnotationElement extends WidgetAnnotationElement {
           event.target.value = elementData.formattedValue;
         }
-        event.target.setSelectionRange(0, 0);
+        event.target.scrollLeft = 0;
```
This *really* needs a good comment, referencing the original issue, since it's less clear exactly what this does than in the previous code; I'd suggest: 
```javascsript
        // Reset the cursor position to the start of the field (issue 12359).
        event.target.scrollLeft = 0;
```

When fixing this, please remember https://github.com/mozilla/pdf.js/wiki/Squashing-Commits

CetinSert committed:
Fix annotation input focus trap regression in Safari

`setSelectionRange(0, 0)` added in https://github.com/mozilla/pdf.js/commit/44b24fcc2936e1d75c249aa161970ab8fa83e4d2 for #12359, required only by Firefox ([bug](https://bugzilla.mozilla.org/show_bug.cgi?id=860329)), causes issues mozilla#13191, mozilla#12592 in Safari.
`scrollLeft = 0` is a fix that breaks the focus trap in Safari while **keeping Firefox behavior same for #12359**.

Snuffleupagus reviewed (changes requested):
As mentioned previously, you *need* to follow https://github.com/mozilla/pdf.js/wiki/Squashing-Commits since we cannot consider landing this in its current state.

Note also that it's not advisable to do development directly against the `master` branch, please see https://github.com/mozilla/pdf.js/wiki/Contributing

cetinsert commented:
@Snuffleupagus great, so this will now take me at least a few more days to my workflow up properly.

cetinsert commented:
@Snuffleupagus see https://github.com/mozilla/pdf.js/pull/13232. Thank you for your prompt and patient guidance!
